src/main/java/com/eu/habbo/habbohotel/rooms/Room.java

```diff
@@ -4873,4 +4877,162 @@ public Collection<RoomUnit> getRoomUnitsAt(RoomTile tile) {
         THashSet<RoomUnit> roomUnits = getRoomUnits();
         return roomUnits.stream().filter(unit -> unit.getCurrentLocation() == tile).collect(Collectors.toSet());
     }
+
+    public List<HabboItem> getAreaHiders(boolean onlyActive) {
+        TIntObjectIterator<HabboItem> iterator = this.roomItems.iterator();
+        List<HabboItem> foundItems = new ArrayList<>();
+        Gson gson = new Gson();
```

Comment:
`gson` should be a global instance, there should be one somewhere in the codebase already (I hope).